src/libraries/System.Net.Http/tests/FunctionalTests/NtAuthTests.FakeServer.cs

```diff
@@ -0,0 +1,140 @@
+
+                // Read next requests and fall-back to loop bellow to process it.
+                request = await connection.ReadRequestDataAsync();
+            }
+
+            bool isAuthenticated = false;
+            do
+            {
+                foreach (HttpHeaderData header in request.Headers)
+                {
+                    if (header.Name == "Authorization")
+                    {
+                        authHeader = header.Value;
+                        break;
+                    }
+                }
+
+                Assert.NotNull(authHeader);
+                var tokens = authHeader.Split(' ', 2, StringSplitOptions.TrimEntries);
+                // Should be type and base64 encoded blob
+                Assert.Equal(2, tokens.Length);
+
+                if (fakeNtlmServer == null)
+                {
+                    fakeNtlmServer = new FakeNtlmServer(s_testCredentialRight) { ForceNegotiateVersion = true };
+                    if (!useNtlm)
+                    {
+                        fakeNegotiateServer = new FakeNegotiateServer(fakeNtlmServer);
+                    }
+                }
+
+                byte[]? outBlob;
+
+                if (fakeNegotiateServer != null)
+                {
+                    outBlob = fakeNegotiateServer.GetOutgoingBlob(Convert.FromBase64String(tokens[1]));
+                    isAuthenticated = fakeNegotiateServer.IsAuthenticated;
+                }
+                else
+                {
+                    outBlob = fakeNtlmServer.GetOutgoingBlob(Convert.FromBase64String(tokens[1]));
+                    isAuthenticated = fakeNtlmServer.IsAuthenticated;
+                }
+
+                if (outBlob != null)
+                {
+                    authHeader = $"WWW-Authenticate: {tokens[0]} {Convert.ToBase64String(outBlob)}\r\n";
+                    await connection.SendResponseAsync(isAuthenticated ? HttpStatusCode.OK : HttpStatusCode.Unauthorized, authHeader);
+                    connection.CompleteRequestProcessing();
+
+                    if (!isAuthenticated)
+                    {
+                        request = await connection.ReadRequestDataAsync();
+                    }
+                }
+            }
+            while (!isAuthenticated);
+
+            await connection.SendResponseAsync(HttpStatusCode.OK);
+        }
+
+        [ConditionalTheory(nameof(IsNtlmInstalled))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task SocketsHttpHandler_FakeServer_Success(bool useNtlm)
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Get, uri);
+                    requestMessage.Version = new Version(1, 1);
+
+                    var handler = new HttpClientHandler() { Credentials = s_testCredentialRight };
```

Comment:
I actually meant to use `SocketsHttpHandler` here and add separate `HttpClientHandler` test later. However, since I already did the mistake, let's wait for the CI run to complete to see how the mobile platforms behave.